modules/cstrike/csx/meta_api.cpp

```diff
@@ -178,14 +178,18 @@ void PlayerPreThink_Post( edict_t *pEntity ) {
 	RETURN_META(MRES_IGNORED);
 }
 
-void ServerDeactivate() {
+void ServerDeactivate() 
+{
 	int i;
-	for( i = 1;i<=gpGlobals->maxClients; ++i){
-		CPlayer *pPlayer = GET_PLAYER_POINTER_I(i);
-		if (pPlayer->rank) pPlayer->Disconnect();
+
+	for( i = 1; i <= gpGlobals->maxClients; ++i)
+	{
+		GET_PLAYER_POINTER_I(i)->Disconnect();
 	}
-	if ( (g_rank.getRankNum() >= (int)csstats_maxsize->value) || ((int)csstats_reset->value == 1 ) ) {
-		CVAR_SET_FLOAT("csstats_reset",0.0);
+
+	if (int(csstats_maxsize->value) <= 0 || g_rank.getRankNum() >= int(csstats_maxsize->value) || int(csstats_reset->value) != 0)
```

Comment:
AFAIK, `int(x)` is not a C-style cast and should be the same as `static_cast<int>(x)`. I think both are fine, sometimes it feels appropriate to use `static_cast`, sometimes not. Well, I guess for consistency, we should probably always use `*_cast`.